### PR TITLE
Make MlflowClient serializable

### PR DESCRIPTION
**Pull request by smurching**
## What changes are proposed in this pull request?

Makes the Java MlflowClient serializable to facilitate use in Spark jobs

## How is this patch tested?
Existing tests, plus tried it out in a Spark job

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Model Registry
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by AveshCSingh — APPROVED** [review]
LGTM

File: mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java [hunk 1]
```diff
@@ -21,7 +22,7 @@
 /**
  * Client to an MLflow Tracking Sever.
  */
-public class MlflowClient {
+public class MlflowClient extends Serialiazable {
```
Comment (aarondav) [inline mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java:25]:
```suggestion
public class MlflowClient implements Serialiazable {
```

File: mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java [hunk 2]
```diff
@@ -21,7 +22,7 @@
 /**
  * Client to an MLflow Tracking Sever.
  */
-public class MlflowClient {
+public class MlflowClient implements Serialiazable {
```
Comment (aarondav) [inline mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java:25]:
```suggestion
public class MlflowClient implements Serializable {
```

**Comment by smurching** [comment]
Lol thanks for the fixes @aarondav, I somehow managed to achieve a nearly 1:1 bug to line-of-code ratio on this PR, and thanks for the review @AveshCSingh, merging :)

**Pull request merged**